base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -679,7 +679,21 @@ protected void updateValidityChecks(ActionsViewTable viewTable) {
      * 
      * @param viewTable The table view to add columns to.
      */
-    protected void addColumns(ActionsViewTable viewTable) {        
+    protected void addColumns(ActionsViewTable viewTable) {
+    // Add line number
+	TableViewerColumn lineNumberColumn = viewTable.createColumn("Line", 
+	        0, 
+	        new DataboundCellLabelProvider<ScriptGeneratorAction>(viewTable.observeProperty("move")) {
+	        @Override
+	        protected String stringFromRow(ScriptGeneratorAction row) {
+	        	return row.getLineNumber().toString();
+	        }
+	        @Override
+            public String getToolTipText(Object element) {
+                return getScriptGenActionToolTipText((ScriptGeneratorAction) element);
+            }
+	    });
```

Comment:
By replacing this with:
```suggestion
	TableViewerColumn lineNumberColumn = viewTable.createColumn("Line", 
	        0, 
	        new SortableObservableMapCellLabelProvider<ScriptGeneratorAction>(viewTable.observeProperty("Line")) {
                
                @Override
                public void update(ViewerCell cell) {
                    
                    for (int i = 0; i < viewTable.table().getItemCount(); i++) {
                        if (cell.getElement().equals(viewTable.viewer().getElementAt(i))) {
                            cell.setText(String.valueOf(i + 1));
                            break;
                        }
                    }
                    
                }

                @Override
                protected String stringFromRow(ScriptGeneratorAction row) {
                    // TODO Auto-generated method stub
                    return null;
                }
            });
```

You can remove the need to keep track of the line number in the back end. It needs a bit of tidying up as we can instead use a `CellLabelProvider` rather than a `SortableObservableMapCellLabelProvider` but I think overall it would be cleaner, what do you think?